### PR TITLE
Open group activity pages style

### PR DESCRIPTION
**Pull request by hmstepanek**
Reformat the side bar on group/user activity pages

Contains minor tweaks to colors, spacing, and adds some additional info (like the day to the date, a Moderators section for open groups, and a creator label next to the user who created the group). This is a fix for https://github.com/hypothesis/product-backlog/issues/490.

image 1
image

List of changes:
- Increase spacing between sections on the sidebar.
- Standardize spacing between section titles and their content as well as between sections.
- Tighten up the grey box around the tag to be closer to the text in order to not create an optical illusion of the spacing between the title of the section and the tag varying from other sections.
- Update grey colors on the users/tag names to be 
         name #.....................to be..............dark-grey lightest-grey
         name # creator.........to be..............dark-grey lightest-grey light-grey
- Level the text in tags and users to be at the same level as the #s (previously the numbers were slightly higher)
- Move group description, dl's, and edit/join links into their own section (this also includes a change from using dl to using p and span elements).
- Move the share/invite info into it's own section also - this just makes it easier to keep everything consistent and in one place.

### CODE REVIEW

**Comment by hmstepanek** [comment]
Design/UX changes requested and approved by @dawa

File: h/templates/activity/search.html.jinja2 [hunk 1]
```diff
@@ -211,174 +211,161 @@
...
-      {{ group_users("Moderators", group.moderators, group.creator) }}
-      <h3 class="group-invite__title">
-        {% trans %}Share group{% endtrans %}
-      </h3>
-      <p class="search-result-sidebar__share-link">
-        {% trans %}Sharing the link lets people view this group:{% endtrans %}
-      </p>
+    {% if user.description %}
+      <div class="search-result-sidebar__div-group-info">
+      {% for paragraph in group.description.split('\n') %}
```
Comment (robertknight) [inline h/templates/activity/search.html.jinja2]: Presumably this should be `user.description`? User profile pages are generating an error on this branch:

screenshot 2018-02-26 12 02 43

It also seems like this markup transform is something that might be useful to extract into a helper function/macro.

File: h/static/styles/partials/_search-result-sidebar.scss [hunk 2]
```diff
@@ -1,12 +1,16 @@
-.search-result-sidebar {
+search-result-sidebar {
```
[the remark this answers is not in the document]
Comment (robertknight) [reply h/static/styles/partials/_search-result-sidebar.scss]: I think this is what is causing a change on user profile pages from this on master:

screenshot 2018-02-26 12 07 10

To this branch:

screenshot 2018-02-26 12 07 30

(Note that I had to correct a reference to `group.description` in order to see this, see other comments).

File: h/templates/activity/search.html.jinja2 [hunk 3]
```diff
@@ -211,163 +211,161 @@
...
-          </dt>
-          <dd class="search-result-sidebar__dd">{{ user.location }}</dd>
-        {% endif -%}
-
-        {%- if user.uri %}
-          <dt class="search-result-sidebar__dt">
-            {% trans %}Link:{% endtrans %}
-          </dt>
-          <dd class="search-result-sidebar__dd">
+      <div class="search-result-sidebar__div-group-info">
```
Comment (robertknight) [inline h/templates/activity/search.html.jinja2]: This styles all contain the term `group-info` but are also used for the user profile sidebar. This suggests to me that we should come up with a more generic (and shorter!) naming scheme for these components.

File: h/templates/activity/search.html.jinja2 [hunk 4]
```diff
@@ -211,163 +211,161 @@
 </section>
 {% endmacro %}
 
-{% macro sidebar(title, description) %}
+{% macro group_sidebar(group, group_edit_url, total, stats) %}
 <aside class="search-result-sidebar {%- if not more_info %} search-result-hide-on-small-screens{% endif %}">
```
Comment (robertknight) [inline h/templates/activity/search.html.jinja2]: I'm struggling a little to follow what is now intentionally different in the top-level layout for the user and group sidebar. Can you clarify?

File: h/static/styles/partials/_search-result-sidebar.scss [hunk 5]
```diff
@@ -19,32 +23,31 @@
 }
 
 .search-result-sidebar__section {
-  margin-bottom: 26px;
+  margin-bottom: 2em;
```
[the remark this answers is not in the document]
Comment (hmstepanek) [reply h/static/styles/partials/_search-result-sidebar.scss]: I wanted it to adjust based on the screen size height. I think that will do what I'm after? Let me know if I'm way off here or if there's a better way.

**Comment by hmstepanek** [comment]
@robertknight I've updated the commit history into smaller more-easy to follow changes. Let me know if you want to break this up anymore (I also addressed/fixed some of the things you pointed out). A couple of these changes we may be able to break out but most of them are kinda hard to be standalone as they affect the final look in terms of margin/alignment/etc. I update the description with a list of changes and also added some pictures of the final review I had with @dawariley where she approved. I haven't looked at this on ie so prob should check that. Also qq: why is the form not around the share/invite link section?

File: h/static/styles/partials/_search-result-sidebar.scss (hunk 5 above)
[the remark this answers is not in the document]
Comment (robertknight) [reply h/static/styles/partials/_search-result-sidebar.scss]: > I wanted it to adjust based on the screen height.

_Screen_ height? _em_ is a unit that represents the [calculated `font-size`](https://developer.mozilla.org/en-US/docs/Web/CSS/length) for an element.

[the remark this answers is not in the document]
Comment (hmstepanek) [reply h/static/styles/partials/_search-result-sidebar.scss]: Ok in light of testing out screen height adjusting this morning (using %) -it really doesn't look the way I intended so I'm just setting this to 26 px which is what the em ultimately converted to.

File: h/templates/activity/search.html.jinja2 (hunk 3 above)
Comment (hmstepanek) [reply h/templates/activity/search.html.jinja2]: I changed this to subsection -hope that's a more acceptable name here.

File: h/static/styles/partials/_search-result-sidebar.scss [hunk 6]
```diff
@@ -69,17 +75,18 @@
   border-radius: 2px;
   margin-bottom: 5px;
   margin-right: 5px;
-  padding: 5px;
+  padding: 1px;
```
[the remark this answers is not in the document]
Comment (hmstepanek) [reply h/static/styles/partials/_search-result-sidebar.scss]: I adjusted the left and right to be slightly larger. My reasoning here is that too wide of a padding results in the text looking tabbed in which visually throws off the spacing consistency between sections.

**Pull request merged**